README.md

```diff
@@ -138,13 +138,26 @@ In method signatures, there should be a space after the scope (`-` or `+` symbol
 ```objc
 - (void)setExampleText:(NSString *)text image:(UIImage *)image;
 ```
+
 ## Variables
 
-Variables should be named as descriptively as possible. Single letter variable names should be avoided except as simple counter variables in `for` loops.
+Variables should be named descriptively, with the variable’s name clearly communicating what the variable _is_ and pertinent information a programmer needs to use that value properly.
+
+**For example:**
+
+* `NSString *title`: It is reasonable to assume a “title” is a string.
+* `NSString *titleHTML`: This indicates a title that may contain HTML which needs parsing for display. _“HTML” is needed for a programmer to use this variable effectively._
+* `NSAttributedString *titleAttributedString`: A title, already formatted for display. _`AttributedString` hints that this value is not just a vanilla title, and adding it could be a reasonable choice depending on context._
+* `NSDate *now`: _No further clarification is needed._
+* `NSDate *lastModifiedDate`: Simply `lastModified` can be ambiguous; depending on context, one could reasonably assume it is one of a few different types.
+* `NSURL *url` vs. `NSString *urlString`: In situations when a value can reasonably be represented by different classes, it is often useful to disambiguate in the variable’s name.
```

Comment:
I couldn't find anything specific about this in [Coding Guidelines for Cocoa](https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/CodingGuidelines/CodingGuidelines.html#//apple_ref/doc/uid/10000146-SW1) or [Programming in Objective-C](https://developer.apple.com/library/ios/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/Introduction/Introduction.html#//apple_ref/doc/uid/TP40011210-CH1-SW1).

Disregarding that, I tend not to begin variable names, even acronyms/initialisms, with capital letters due to potential for visual confusion with constants (which traditionally begin with at least one capital) and classes (especially in Objective-C, where class names begin with multiple capital letters).

Say I encounter the symbol `URLString` further down in the code. After a few seconds of reading in-context, it should become clear that it's a variable, not a class, even though that name follows class-naming conventions exactly. But that few seconds is a mental distraction, and a [context switch](http://en.wikipedia.org/wiki/Context_switch#Performance), from what I was trying to accomplish.

I will run into the same problem with constants, but it'll typically be harder to distinguish, since a constant's type is so similar to the type of a similar variable.